### PR TITLE
Soft purge

### PR DESCRIPTION
**Pull request by simonwistow**
Allow Soft Purging of URLs and Surrogate Keys.

Closes issue #16

Also switch to the newer form of PURGE-ing

### CODE REVIEW

File: lib/Net/Fastly.pm [hunk 1]
```diff
@@ -203,15 +203,34 @@ sub commands {
...
 }
 
-=head2 purge <path>
+=head2 purge <path> [soft]
 
 Purge the specified path from your cache.
 
+You can optionally pass in a true value to enable "soft" purging e.g
+
+    $fastly->purge($url, 1);
```
Comment (ranguard) [inline lib/Net/Fastly.pm]: I think this should be `$service->` ?

as $fastly is the client in the rest of the docs

File: lib/Net/Fastly/Service.pm [hunk 2]
```diff
@@ -97,16 +97,23 @@ sub purge_all {
 
 Purge anything with the specific key from the given service. Requires an API key.
 
+You can optionally pass in a true value to enable "soft" purging e.g
+
+    $fastly->purge_by_key($key, 1);
```
Comment (ranguard) [inline lib/Net/Fastly/Service.pm]: same with this

File: lib/Net/Fastly.pm (hunk 1 above)
Comment (ranguard) [inline lib/Net/Fastly.pm]: Oh, actually, think it's only purge_by_key that's `service` based, sorry read that code earlier and saw the issue, assumed same here :)

**Comment by miyagawa** [comment]
:+1:

**Comment by ranguard** [comment]
Merge and deploy :) ?

**Comment by simonwistow** [comment]
@ranguard I'm doing some other refactoring, will merge that too and deploy

**Comment by ranguard** [comment]
:+1:  - thanks

**Comment by ranguard** [comment]
@simonwistow please could you release, I'd really like to start using this, I don't need the refactoring

**Comment by simonwistow** [comment]
Released as v1.04 to CPAN

**Pull request merged**